src/graph/toRest.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * method to convert graph responses to REST responses
+ */
+
+import {
+  GetSavedItemsQuery,
+  Imageness,
+  Videoness,
+} from '../generated/graphql/types';
+import {
+  GraphItem,
+  GraphSavedItem,
+  GraphSavedItemEdge,
+  ListItemObject,
+  RestResponse,
+} from './types';
+
+/**
+ * process if the item fields are populated. if its pendingItem, we just return null
+ * @param savedItem
+ */
+
+function convertHasImage(imageStatus: Imageness) {
+  switch (imageStatus) {
+    case Imageness.IsImage:
+    case Imageness.HasImages:
+      return '1';
+    case Imageness.NoImages:
+      return '0';
+    default:
+      return '0';
+  }
+}
+
+function convertHasVideo(videoStatus: Videoness) {
+  switch (videoStatus) {
+    case Videoness.HasVideos:
+    case Videoness.IsVideo:
+      return '1';
+    case Videoness.NoVideos:
+      return '0';
+    default:
+      return '0';
+  }
+}
+
+/**
+ * converts list to map
+ * @param input list of entities
+ * @param key key to assign the entity
+ */
+function listToMap<T>(input: T[], key: string): { [key: string]: T } {
+  return input.reduce((map, item) => {
+    map[item[key]] = item;
+    return map;
+  }, {});
+}
+
+/**
+ * converts saves.Items to REST response.
+ * sets the listItem as null for PendingItem
+ * @param savedItemEdge savedItem edge from the graph
+ */
+const reduceItem = (savedItemEdge: GraphSavedItemEdge): ListItemObject => {
```

Comment:
same here, we probably won't have the relay connection pattern for offset pagination but we can just refactor later